micro-rdk/src/esp32/certificate.rs

```diff
@@ -28,3 +49,277 @@ impl Certificate for WebRtcCertificate {
+        let not_after = CString::new(self.not_after.clone()).unwrap();
+
+        unsafe {
+            MbedTLSError::to_unit_result(mbedtls_x509write_crt_set_validity(
+                &mut self.crt_context as *mut mbedtls_x509write_cert,
+                not_before.as_ptr(),
+                not_after.as_ptr(),
+            ))
+        }?;
+
+        let mut work_buffer = vec![0_u8; 2048];
+
+        let ret = unsafe {
+            MbedTLSError::to_result(mbedtls_x509write_crt_der(
+                &mut self.crt_context as *mut mbedtls_x509write_cert,
+                work_buffer.as_mut_ptr(),
+                2048,
+                Some(mbedtls_ctr_drbg_random),
+                &mut self.drbg_context as *mut _ as *mut c_void,
+            ))
+        }?;
+        let der_certificate = work_buffer[2048 - ret as usize..].to_vec();
+        assert_eq!(der_certificate.len(), ret as usize);
+
+        unsafe {
+            esp_sha(
+                SHA_TYPE_SHA2_256,
+                der_certificate.as_ptr(),
+                der_certificate.len(),
+                work_buffer.as_mut_ptr(),
+            )
+        };
+        let der_fp = work_buffer
+            .iter()
+            .take(32)
+            .map(|b| format!("{:02X}", b))
+            .collect::<Vec<String>>()
+            .join(":");
+        assert_eq!(der_fp.len(), 95);
+
+        let ret = unsafe {
+            MbedTLSError::to_result(mbedtls_pk_write_key_der(
+                &mut self.kp_context as *mut _,
+                work_buffer.as_mut_ptr(),
+                2048,
+            ))
+        }?;
+
+        let der_kp = work_buffer[2048 - ret as usize..].to_vec();
+        assert_eq!(der_kp.len(), ret as usize);
+
+        let fingerprint = Fingerprint::new("sha-256".to_owned(), der_fp);
+
+        Ok(WebRtcCertificate {
+            serialized_der: der_certificate,
+            key_pair: der_kp,
+            fingerprint,
+        })
+    }
+}
+
+impl Default for GeneratedWebRtcCertificateBuilder {
+    fn default() -> Self {
+        let drbg_context = unsafe {
+            let mut drbg_context = MaybeUninit::uninit();
+            mbedtls_ctr_drbg_init(drbg_context.as_mut_ptr());
+            drbg_context.assume_init()
+        };
+        let entropy = unsafe {
+            let mut entropy = MaybeUninit::uninit();
+            mbedtls_entropy_init(entropy.as_mut_ptr());
+            entropy.assume_init()
+        };
+        let kp_context = unsafe {
+            let mut kp_context = MaybeUninit::uninit();
+            mbedtls_pk_init(kp_context.as_mut_ptr());
+            kp_context.assume_init()
+        };
+        let crt_context = unsafe {
+            let mut crt_context = MaybeUninit::uninit();
+            mbedtls_x509write_crt_init(crt_context.as_mut_ptr());
+            crt_context.assume_init()
+        };
+
+        Self {
+            oid_cn: String::from("VIAM Webrtc"),
+            oid_c: String::from("US"),
+            oid_o: String::from("VIAM"),
```

Comment:
Is there some existing art that indicates these specific values? Do they need to conform with anything specific?

Reply:
No in most cases they recommend to set them so some sane values but its doesn't look like it's super critical since it is self signed.